### PR TITLE
mach: Do not use unstable rust for `rustfmt`

### PR DESCRIPTION
**Pull request by mrobinson**
We can use stable rust if we pass the unstable configuration as
command-line arguments to rustfmt itself. This prevents needing to
install an unstable rust toolchain.

The one downside here is that it doesn't seem that "ignore" is
supported so we have to start formatting the files in "third_party."
This shouldn't be a huge issue because we don't plan to check much more
rust code into those directories.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes do not require tests because they just adjust the `rustfmt` call.

### CODE REVIEW

File: python/servo/testing_commands.py [hunk 1]
```diff
@@ -39,6 +39,16 @@
 WEB_PLATFORM_TESTS_PATH = os.path.join("tests", "wpt", "tests")
 SERVO_TESTS_PATH = os.path.join("tests", "wpt", "mozilla", "tests")
 
+# Servo depends on several `rustfmt` that are unstable. These are still supported by
+# stable `rustfmt if they are passed as command-line arguments. These are the
+# arguments we sue.
```
Comment (mukilan) [inline python/servo/testing_commands.py:44]:
```suggestion
# arguments we use
```

File: python/servo/testing_commands.py [hunk 2]
```diff
@@ -39,6 +39,16 @@
 WEB_PLATFORM_TESTS_PATH = os.path.join("tests", "wpt", "tests")
 SERVO_TESTS_PATH = os.path.join("tests", "wpt", "mozilla", "tests")
 
+# Servo depends on several `rustfmt` that are unstable. These are still supported by
+# stable `rustfmt if they are passed as command-line arguments. These are the
```
Comment (mukilan) [inline python/servo/testing_commands.py:43]:
```suggestion
# stable `rustfmt` if they are passed as command-line arguments. These are the
```

File: python/servo/testing_commands.py [hunk 3]
```diff
@@ -39,6 +39,16 @@
 WEB_PLATFORM_TESTS_PATH = os.path.join("tests", "wpt", "tests")
 SERVO_TESTS_PATH = os.path.join("tests", "wpt", "mozilla", "tests")
 
+# Servo depends on several `rustfmt` that are unstable. These are still supported by
```
Comment (mukilan) [inline python/servo/testing_commands.py:42]:
```suggestion
# Servo depends on several `rustfmt` arguments that are unstable. These are still supported by
```

**Comment by mrobinson** [comment]
@mukilan Thanks for the review. I've updated the PR fixing the typos.

**Review by mukilan — APPROVED** [review]

**Pull request merged**